src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2494,42 +2551,43 @@ private void addRelatedInterfaces(ObjectType instance, Set<ObjectType> interface
       }
     }
 
-    private void visitProperties(ObjectType objType, boolean isStatic) {
-      visitProperties(
-          objType,
-          isStatic,
-          Collections.<String>emptySet(),
-          Collections.<String>emptySet(),
-          Collections.<String>emptyList());
+    /**
+     * Emits the given set of properties.
+     *
+     * @param isInNamespace if true, emit the properties in a form suitable for namespace members
+     *     (e.g. with "var" and "function" prefixes). If false, emit suitable for properties
+     *     declared in a class or interface.
+     */
+    private void visitStaticProperties(
+        ObjectType objType, Set<String> propNames, boolean isInNamespace) {
+      for (String propName : propNames) {
+        if (isInNamespace) {
+          JSType propType = objType.getPropertyType(propName);
+          if (propType.toMaybeFunctionType() == null) {
+            emit("var");
+          } else {
+            emit("function");
+          }
+        }
+        visitProperty(
+            propName,
+            objType,
+            !isInNamespace,
+            false,
+            isInNamespace,
+            Collections.<String>emptyList());
+      }
     }
 
-    private void visitProperties(
+    private void visitInstanceProperties(
```

Comment:
it turns out this was only ever called directly with isStatic=false, so I refactored to split the two methods and call `visitProperty` directly. That allowed passing down the `propNames` for the static case, which in turn allows us to check conveniently beforehand if there are any properties we need to emit, so that we can avoid empty `namespace` declarations for interfaces that do not have static props. I also had to pull up some conditional checks into `getTypePropertyNamesToEmit`, I think by and large it's a bit more sensible now.

Reply:
This function is getting so small, it almost feels like it should be inlined, but I guess it is nice to keeps for symmetry with visitStaticProperties  which has more meat.

Reply:
I like the symmetry, without it it's a bit unclear why this code path exists at all.